### PR TITLE
PowerDistributor should not block user command.

### LINKED ISSUE #91: PowerDistributor blocks user commands if all components are considered as broken
### What happened?

PowerDistributor blocks user commands if all components are considered as broken.

### What did you expect instead?

Better detection if components is broken or not: https://github.com/frequenz-floss/frequenz-sdk-python/issues/33
As a workaround in this PR:
If all batteries are considered as broken, then user should anyway be able to use them.

### Affected version(s)

v0.12.0

### Affected part(s)

Battery power distribution (part:power-distribution)

### Extra information

_No response_

### PR DESCRIPTION
**Pull request by ela-kotulska-frequenz**
This is workaround for the bug.
Previously if all components were considered as broken, then all user commands were blocked for 30 seconds. With this workaround if all components are considered as broken then we will try to use them anyway.

### CODE REVIEW

File: tests/actor/test_power_distributing.py [hunk 1]
```diff
@@ -738,3 +740,43 @@ async def test_power_distributor_stale_all_components_message(
...
+            20 * [datetime.fromisoformat("2001-01-01T00:00:31+00:00")]
+        )
+
+        datetime_mock.now.side_effect = expected_datetime
+
+        # After 30 seconds components should be considered as working
+        broken = _BrokenComponents(30)
+
+        for i in range(3):
+            broken.mark_as_broken(i)
```
Comment (sahas-subramanian-frequenz) [inline tests/actor/test_power_distributing.py]: interesting,  pylint always complains to me when I use single letter variable names.

File: src/frequenz/sdk/actor/power_distributing.py [hunk 2]
```diff
@@ -91,6 +91,34 @@ def update_retry(self, timeout_sec: float) -> None:
...
+        """Get subset of batteries that are not marked as broken.
+
+        If all given batteries are broken, then mark them as working and return them.
+        This is temporary workaround to not block user command.
+
+        Args:
+            components_ids (Set[int]): set of component ids
+
+        Returns:
+            Set[int]: subset of given components_ids with working components.
```
Comment (sahas-subramanian-frequenz) [inline src/frequenz/sdk/actor/power_distributing.py]: we are not specifying types for args or returns anymore.

File: tests/actor/test_power_distributing.py (hunk 1 above)
Comment (ela-kotulska-frequenz) [reply tests/actor/test_power_distributing.py]: Done

File: src/frequenz/sdk/actor/power_distributing.py (hunk 2 above)
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/actor/power_distributing.py]: True, I have to set it in VsCode.

Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/actor/power_distributing.py]: Done

File: tests/actor/test_power_distributing.py (hunk 1 above)
Comment (ela-kotulska-frequenz) [reply tests/actor/test_power_distributing.py]: Maybe it is related with python version number?
What python you are using?

Comment (sahas-subramanian-frequenz) [reply tests/actor/test_power_distributing.py]: it is fine,  I was just worried that pylint might not be running on the test files for some reason and we are not catching other issues as well.  I will test it separately.

**Review by sahas-subramanian-frequenz — APPROVED** [review]

File: tests/actor/test_power_distributing.py (hunk 1 above)
Comment (leandro-lucarella-frequenz) [reply tests/actor/test_power_distributing.py]: it probably have an exception for `for` loops.

**Pull request merged**